pkg/reconciler/route/reconcile_resources.go

```diff
@@ -123,37 +126,101 @@ func (c *Reconciler) reconcileClusterIngress(
 	return clusterIngress, err
 }
 
-func (c *Reconciler) reconcilePlaceholderService(ctx context.Context, route *v1alpha1.Route, ingress *netv1alpha1.ClusterIngress) error {
+func (c *Reconciler) deleteServices(namespace string, serviceNames sets.String) error {
+	for _, serviceName := range serviceNames.List() {
+		if err := c.KubeClientSet.CoreV1().Services(namespace).Delete(serviceName, nil); err != nil {
+			return err
```

Comment:
Do you want to add some logging here in the error case?

Reply:
Done